src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyResultSet.java

```diff
@@ -519,13 +520,32 @@ public void regenerateVariationEvents(int maxMnpDistance) {
     }
 
     /**
-     * Get all of the VariantContexts in the event maps for all haplotypes, sorted by their start position
+     * Get all of the VariantContexts in the event maps for all haplotypes, sorted by their start position and then arbitrarily by indel length followed by bases
      * @param haplotypes the set of haplotypes to grab the VCs from
      * @return a sorted set of variant contexts
      */
     private static SortedSet<VariantContext> getAllVariantContexts( final List<Haplotype> haplotypes ) {
         // Using the cigar from each called haplotype figure out what events need to be written out in a VCF file
-        final TreeSet<VariantContext> vcs = new TreeSet<>(Comparator.comparingInt(VariantContext::getStart));
+        final TreeSet<VariantContext> vcs = new TreeSet<>(
+                Comparator.comparingInt(VariantContext::getStart)
+                        // Decide arbitrarily so as not to accidentally throw away overlapping variants
+                .thenComparing(new Comparator<VariantContext>() {
+                    @Override
+                    public int compare(VariantContext v1, VariantContext v2) {
+                        List<Allele> v1Alleles = v1.getAlleles();
+                        List<Allele> v2Alleles = v2.getAlleles();
+                        Utils.validate(v1Alleles.size() == 2, () -> "Error Haplotype event map Variant Context has too many alleles: "+v1);
```

Comment:
This feels both too `private` for the validation to be necessary and too deep for it to be helpful.  Could you find a higher entry point to check that the variant contexts in the `EventMap` are all biallelic, if one does not exist already?

Reply:
I pushed this up into event map construction.